src/ConnectionCard/ConnectionCard.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { Badge } from '../Badge';
+import { Price } from '../Price';
+import ConnectionCardRow from './ConnectionCardRow';
+import TripSector from './TripSector';
+import BadgesContainer from './BadgesContainer';
+import type { BadgeProps } from '../Badge/BadgeTypes';
+import type { PriceProps } from '../Price/PriceTypes';
+import type { TripSectorProps } from './ConnectionCardTypes';
+import StyleSheet from '../PlatformStyleSheet';
+
+type TripSectorWithId = { ...TripSectorProps, id: number };
+type BadgeWithId = { ...BadgeProps, id: number };
+
+type Props = {|
+  padding?: boolean,
+  wayForth: Array<TripSectorWithId>,
+  wayBack?: Array<TripSectorWithId>,
+  price: PriceProps,
+  duration?: string,
+  badges?: Array<BadgeWithId>,
+|};
+
+const renderTripSector = way =>
+  way.map(connection => (
+    <TripSector
+      key={connection.id}
+      arrival={connection.arrival}
+      arrivalTime={connection.arrivalTime}
+      carrier={connection.carrier}
+      tripDate={connection.tripDate}
+      departure={connection.departure}
+      departureTime={connection.departureTime}
+      duration={connection.duration}
+    />
+  ));
+
+const renderBadges = (badges: Array<BadgeWithId>) =>
+  badges.map(badge => (
+    <Badge key={badge.id} type={badge.type} style={styles.badge}>
+      {badge.children}
+    </Badge>
+  ));
+
+export default function ConnectionCard({
+  padding = true,
+  wayForth,
+  wayBack,
+  price,
+  duration,
+  badges,
+}: Props) {
+  const hasBadges = badges && badges.length > 0;
+  const hasWayBack = wayBack && wayBack.length > 0;
+  return (
+    <View style={[styles.container, { paddingHorizontal: padding ? 10 : 0 }]}>
+      <ConnectionCardRow>{renderTripSector(wayForth)}</ConnectionCardRow>
+      {hasWayBack && (
+        <React.Fragment>
+          {duration && (
+            <ConnectionCardRow>
+              <View style={styles.leftShift}>
+                <Badge type="neutral">Returns in {duration}</Badge>
+              </View>
+            </ConnectionCardRow>
+          )}
+          <ConnectionCardRow>
+            {wayBack && renderTripSector(wayBack)}
+          </ConnectionCardRow>
+        </React.Fragment>
+      )}
+      <ConnectionCardRow style={styles.lastRow}>
+        {hasBadges && (
+          <BadgesContainer>{badges && renderBadges(badges)}</BadgesContainer>
```

Comment:
I thinking about this part of the code
```
 <BadgesContainer>{badges && renderBadges(badges)}</BadgesContainer>
```

maby for better readability, we can do something like this in the `renderBadges` function:
(same for`renderTripSector` )
```
const renderBadges = badges => {
  if (!badges) return undefined;
   badges.map(badge => (
....
```
So main render code will be cleaner.
What do you think about that?

Reply:
It's definitely more readable. And what about this one? I think it's even easier.

```js
const renderBadges = badges =>
  badges &&
  badges.map(badge => (
    <Badge key={badge.id} type={badge.type} style={styles.badge}>
      {badge.children}
    </Badge>
  ));

...

<BadgesContainer>{renderBadges(badges)}</BadgesContainer>
```

Reply:
For me are both OK, so choose one.